Deleted the previos part of code and changes have been done according…

MarijaBorisova opened the pull request:
… to Mentor's request: manually input of code and random separately. Manually input- 1 iteration finished, Random- without logic, only print

MarijaBorisovaAccenture committed:
Deleted the previos part of code and changes have been done according to Mentor's request: manually input of code and random separately. Manually input- 1 iteration finished, Random- without logic, only print

MarijaBorisovaAccenture committed:
Random number Console 1 iteration is finished

MarijaBorisovaAccenture committed:
Deleted the first ex before Field 0

MarijaBorisovaAccenture committed:
the task: clean every iteration un input print speed

EdgarsBondars commented on GameOfLifeConsoleRandom/GameSeed.cs:
```diff
@@ -0,0 +1,176 @@
+        int changesCounter; //counter that holds the number of changes that have been iterated
+
+
+        //public int ChangesCounter //property to allow the user to get the current generation
```
A lot of commented out code.

MarijaBorisova replied:
Done according to the requested rules underlined by mentor.
/// before the methods
// before the code line, not in line.
to delete unnecessary codelines

EdgarsBondars commented on GameOfLifeConsole/Program.cs:
```diff
@@ -0,0 +1,70 @@
+﻿namespace GameOfLifeConsole
```
Too many projects. Evaluate which one you will stick with and remove other ones.

MarijaBorisova replied:
Done. Two first ones were in the beginning and saved in masterbranch. So then the decision was to create the new one, in which to save the correct projects according to mentor's request. "GameOfLifeConsole" and "GameOfLifeConsoleRandom"- the right one, which should be updated and revised.

EdgarsBondars commented on GameOfLifeConsoleRandom/GameSeed.cs:
```diff
@@ -0,0 +1,176 @@
+        //    get { return changesCounter; }
+        //}
+
+        public GameSeed(int[,] newField) //const. creates the field of cells by cloning the array
```
Add summary comments for methods.
Write '///' in line before method and visual studio will generate a summary method template.

MarijaBorisova replied:
Done

EdgarsBondars commented on GameOfLifeConsoleRandom/GameSeed.cs:
```diff
@@ -0,0 +1,176 @@
+
+
+        //passes the coordinates of the cell and count the number of neighbours the cell has
+        private int Neighbours(int x, int y)
```
Use two for loops for getting info about alive neighbors. It will be a lot shorter.

MarijaBorisova replied:
Done.

EdgarsBondars commented on GameOfLifeConsoleRandom/GameSeed.cs:
```diff
@@ -0,0 +1,176 @@
+        //passes the coordinates of the cell and count the number of neighbours the cell has
+        private int Neighbours(int x, int y)
+        {
+            int count = 0;//variable to hold the number of neighbours the cell has
```
In some places you have comment line before code and some places same line with code.
I'd suggest you to writing comments line before code.

MarijaBorisova replied:
Done

EdgarsBondars commented on GameOfLifeConsoleRandom/GameSeed.cs:
```diff
@@ -0,0 +1,176 @@
+
+
+        //passes the coordinates of the cell and count the number of neighbours the cell has
+        private int Neighbours(int x, int y)
```
I'd suggest to rename this method to something like GetNeighbourCount as it gives more information on what this method actually does. Neighbours can be confused for something else.

MarijaBorisova replied:
Done. Changed onto NeighboursCount.

EdgarsBondars commented on GameOfLifeConsoleRandom/GameSeed.cs:
```diff
@@ -0,0 +1,176 @@
+                    //	   (Neighbours(x, y) < 2 || Neighbours(x, y) > 3))// in the next generation will not be alive cell
+                    //                   newGameField[y, x] = 1; //if false (0)- it will die
+                    if (gameField[y, x] == 1 &&
+                           (Neighbours(x, y) == 2 || Neighbours(x, y) == 3))// in the next generation will not be alive cell
```
You can save result of method Neighbors in some variable. Otherwise in some scenarios this method is called twice.

MarijaBorisova replied:
Done.

MarijaBorisovaAccenture committed:
Some changes have been done

MarijaBorisovaAccenture committed:
Deleted unnecessary projects

MarijaBorisovaAccenture committed:
Changes have been done taking an account mentor's comments

MarijaBorisovaAccenture committed:
RandomConsole some changes

MarijaBorisovaAccenture committed:
During the testing, the mistake in NeighboursCount has been found. Resolved

EdgarsBondars commented on GameOfLifeConsoleRandom/Program.cs:
```diff
@@ -0,0 +1,78 @@
+                    Console.ReadLine();
+                    continue;
+                }
+                
```
Empty lines are unnecessary here.

MarijaBorisova replied:
The code has been cleaned up.

EdgarsBondars commented on GameOfLifeConsoleRandom/Program.cs:
```diff
@@ -0,0 +1,78 @@
+                        {
+                            Console.ReadLine();
+                            //break;
+                        }
```
Empty lines after closing brackets are unnecessary here.

MarijaBorisova replied:
Done.

EdgarsBondars commented on GameOfLifeConsoleRandom/Program.cs:
```diff
@@ -0,0 +1,78 @@
+                        else
+                        {
+                            Console.ReadLine();
+                            //break;
```
Commented out code.

MarijaBorisova replied:
Done.

EdgarsBondars commented on GameOfLifeConsoleRandom/Program.cs:
```diff
@@ -0,0 +1,78 @@
+                    {
+
+                        Console.WriteLine();
+                        //Console.WriteLine("Field {0}", gameSeed.ChangesCounter);
```
Remove commented out code.

MarijaBorisova replied:
Done.

EdgarsBondars commented on GameOfLifeConsoleRandom/Program.cs:
```diff
@@ -0,0 +1,78 @@
+                    Console.WriteLine("Field 0");
+                    gameSeed.DrawField();
+                    Console.WriteLine();
+
```
Too many empty lines.

MarijaBorisova replied:
Should not I leave one empty line between the code line, between the code part, in which the method to the object of call is implied and between the print method? Will not it be such kind of mix?

EdgarsBondars commented on GameOfLifeConsoleRandom/Program.cs:
```diff
@@ -0,0 +1,78 @@
+
+                    int[,] gameField = new int[row, column];
+                    GameSeed gameSeed = new GameSeed(gameField);
+
```
Too many empty lines.

EdgarsBondars commented on GameOfLifeConsoleRandom/Program.cs:
```diff
@@ -0,0 +1,78 @@
+                    //Console.Write("The quantity of cells:\t ");
+                    //int density = int.Parse(Console.ReadLine());
+
+
```
Too many empty lines.

MarijaBorisova replied:
Done

EdgarsBondars commented on GameOfLifeConsoleRandom/GameSeed.cs:
```diff
@@ -0,0 +1,124 @@
+                for (int x = 0; x < row; x++)
+                {
+                    int neighboursNumber = NeighboursCount(x, y);
+                    //if the cell is dead and has three neighbours.
```
Keep the comments in one style like - 
```
// Some comment.
````
One space after slashes, sentence starts with big character, and dot when sentence is completed.
Make sure every comment is like this in this file.

MarijaBorisova replied:
Done.

EdgarsBondars commented on GameOfLifeConsoleRandom/GameSeed.cs:
```diff
@@ -0,0 +1,124 @@
+            return count;
+        }
+
+        public void NewCellGeneration()
```
Summary comment missing.

MarijaBorisova replied:
Done.

EdgarsBondars commented on GameOfLifeConsoleRandom/GameSeed.cs:
```diff
@@ -0,0 +1,124 @@
+        /// <summary>
+        /// passes the coordinates of the cell and count the number of neighbours the cell has
+        /// </summary>
+        /// <param name="x"></param>
```
Add comments for parameters and returns section as well.
Like -
```
/// <summary>
/// Passes the coordinates of the cell and count the number of neighbours the cell has.
/// </summary>
/// <param name="x">X coordinate.</param>
/// <param name="y">Y coordinate.</param>
/// <returns>Alive neighbor count.</returns>
```

Add it for other method summaries as well.

MarijaBorisova replied:
Done.

EdgarsBondars commented on GameOfLifeConsoleRandom/GameSeed.cs:
```diff
@@ -0,0 +1,124 @@
+            column = gameField.GetLength(0);
+            //creates an empty field to store the next changed field
+            changedField = new int[column, row];
+    }
```
Padding is off. 
Should be 2 tabs or 8 spaces.

MarijaBorisova replied:
Do you mean I should not use " ", but /t ?

MarijaBorisova replied:
I did it like " ".

EdgarsBondars commented on GameOfLifeConsoleRandom/GameSeed.cs:
```diff
@@ -0,0 +1,124 @@
+                Console.WriteLine();
+            }
+            Console.WriteLine();
+        }
```
Between methods leave one space in between.
Check other places too.

MarijaBorisova replied:
Done.

EdgarsBondars commented on GameOfLifeConsoleRandom/Program.cs:
```diff
@@ -0,0 +1,78 @@
+﻿
+using System;
```
No need for empty line before using's, one empty line after using's.

MarijaBorisova replied:
Done.

EdgarsBondars commented on GameOfLifeConsoleRandom/GameSeed.cs:
```diff
@@ -0,0 +1,124 @@
+        /// <param name="x"></param>
+        /// <param name="y"></param>
+        /// <returns></returns>
+        public int NeighboursCount(int x, int y)
```
Have you tested if everything works as expected after changes to the logic?
I think there is one little thing missing in this method for it to work properly :P

MarijaBorisova replied:
Thank you for the comment. I corrected. Please, check it.

EdgarsBondars commented on GameOfLifeConsole/Program.cs:
```diff
@@ -0,0 +1,49 @@
+            uint MaxRuns = 10;
+            int runs = 0;
+
+            int[,] seed = new int[,]
```
So you don't have to maintain multiple projects maybe you can combine them by having some kind of selector of game kind that user selects in console.
For example when starting game user has option to select whether he wants to see blinker or randomly generated field.

MarijaBorisova replied:
Done, using switch operator. the second project has been deleted. The main class with the logic game has been remained and UI for random and manual has been input in console.

EdgarsBondars commented on GameOfLife/Program.cs:
```diff
@@ -1,2 +1,12 @@
+	class Program
+	{
+		static void Main(string[] args)
+		{
```
This project seems to be doing nothing. Delete it?

MarijaBorisova replied:
Done. Deleted the file from the folder

EdgarsBondars commented on GameOfLifeConsoleRandom/GameSeed.cs:
```diff
@@ -0,0 +1,124 @@
+            //creates an empty field to store the next changed field
+            changedField = new int[column, row];
+    }
+        public void DrawField()
```
Summary comment missing.

MarijaBorisova replied:
Done.

MarijaBorisovaAccenture committed:
Changes have been done acc. to the mentor's requirements in pull requests. Selector has not been done.

MarijaBorisovaAccenture committed:
Switch selector on manual and Random input

EdgarsBondars commented on GameOfLifeConsole/Program.cs:
```diff
@@ -0,0 +1,136 @@
+                            gameSeedRandom.DrawField();
+                            Console.WriteLine();
+
+                            while (gameSeedRandom.AliveCells() > 0 && runs++ < MaxRuns)
```
This same exact while loop repeats itself two times in this file.
Try to refactor this so it's here only one.

EdgarsBondars replied:
Little hint so you don't go in wrong direction - you don't need new methods for that.

MarijaBorisova replied:
Done.

EdgarsBondars commented on GameOfLifeConsoleRandom/Program.cs:
```diff
@@ -0,0 +1,60 @@
+﻿namespace GameOfLifeConsoleRandom
```
This project is obsolete now. You have project that has this one's functionality. you can delete it from repository. :)

MarijaBorisova replied:
Done.

EdgarsBondars commented on GeneralConsoleGame/Program.cs:
```diff
@@ -0,0 +1,11 @@
+﻿namespace GeneralConsoleGame
```
This project is obsolete now. You have project that has this one's functionality. you can delete it from repository. :)

MarijaBorisova replied:
Done. Thank you.

MarijaBorisovaAccenture committed:
Acc to mentor's review two identical code parts have been input in one.

MarijaBorisovaAccenture committed:
Bug ir recognized and resolved: no clear action, everytime should be rerun. Resolved. Another symbol input- the error is shown. Random variant also works properly.

MarijaBorisovaAccenture committed:
Add private modifiers to the fields in GameSeed class.
Started the second interation: count of iteration added, file saving and restore.
